### PR TITLE
Add sns topic and subscription sources

### PR DESCRIPTION
**Pull request by tomasdembelli**
Related to #392

### CODE REVIEW

File: sources/sns/subscription.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// If topic is not healthy, subscription will not work
+				In: true,
+				// Subscription won't affect the topic
+				Out: false,
+			},
+		})
+	}
```
Comment (dylanratcliffe) [inline sources/sns/subscription.go:73]: We should also be able to link `SubscriptionRoleArn` to an IAM role if it exists

File: sources/sns/topic.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		return nil, err
+	}
+
+	item := &sdp.Item{
+		Type:            "sns-topic",
+		UniqueAttribute: "topicArn",
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            tagsToMap(resourceTags),
+	}
```
Comment (dylanratcliffe) [inline sources/sns/topic.go:47]: We could also link these via the `KmsMasterKeyId` to a KMS key. The source doesn't exist yet but it should be called `kms-key` as per the naming convention so we should be able to do a GET using the ID

**Review by dylanratcliffe — CHANGES_REQUESTED** [review]
Two additional links I think we could make, otherwise excellent

File: sources/sns/subscription.go (hunk 1 above)
Comment (tomasdembelli) [reply sources/sns/subscription.go:73]: https://github.com/overmindtech/aws-source/pull/415/commits/e47b3d93b03a1e3f4120730ba5979e5de54d54f3

File: sources/sns/topic.go (hunk 2 above)
Comment (tomasdembelli) [reply sources/sns/topic.go:47]: Great catch, thx!

https://github.com/overmindtech/aws-source/pull/415/commits/b9a458cfa0da9a710053b306833a8055bbffe1b4

**Review by dylanratcliffe — APPROVED** [review]
Nice, I'm going to make just one change before merging which is to make sure that if we're missing a `subscriptionRoleArn` we don't fail the whole mapping: https://github.com/overmindtech/aws-source/pull/420/commits/1da40f4002fb54a6b75362a4e8bc997aa0aa1596

**Pull request merged**